packages/web3-utils/src/utils.js

```diff
@@ -232,9 +232,19 @@ var hexToNumber = function (value) {
     if (typeof value === 'string' && !isHexStrict(value)) {
         throw new Error('Given value "'+value+'" is not a valid hex string.');
     }
+    
 
-    return toBN(value).toNumber();
-};
+    const [negative, hexValue] = String(value).startsWith("-")
+        ? [true, value.slice(1)]
+        : [false, value];
+    const num = toBN(hexValue);
+
+    if (num > Number.MAX_SAFE_INTEGER) {
+        return negative ? -num : num;
+    }
+
+    return negative ? -1 * num.toNumber() : num.toNumber();
```

Comment:
Try this out, hopefully will solve the problem. 

```suggestion
    return 1 * toBN(value); 
```